lib/async_hooks.js

```diff
@@ -366,26 +353,24 @@ function emitBeforeS(asyncId, triggerId = asyncId) {
 
   pushAsyncIds(asyncId, triggerId);
 
-  if (async_hook_fields[kBefore] === 0) {
-    return;
-  }
-
   emitBeforeN(asyncId);
 }
 
 
 // Called from native. The asyncId stack handling is taken care of there before
 // this is called.
 function emitAfterN(asyncId) {
-  if (async_hook_fields[kAfter] > 0) {
-    processing_hook = true;
+  processing_hook = true;
```

Comment:
Note for future change: I'd like to include a `macros.py` similar to `deps/v8/src/js/macros.py` for Debug builds. except it replaces comments instead of normal expressions.

In the code you'll find `// CHECK()` comments that were meant as reminders of important bits to check, but not important enough to check in a Release build. Though the difference would be that
```js
// CHECK(expr)
```
is replaced with something like
```js
assert.ok(expr);
```
But only for debug builds.

Point here would be to include the following here:
```js
// CHECK(async_hook_fields[kAfter] > 0)
```
as a sanity check for the future in case a change is made that invalidates your comment of

> The idea is that the native code doesn’t bother calling `emitAfterN` if the field is 0, so checking it here is pointless for the common case of a call from C++.